pkg/builder/statefulset_builder.go

```diff
@@ -157,12 +158,13 @@ func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.Per
 }
 
 func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (autoMount *bool, serviceAccount string) {
```

Comment:
```suggestion
func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (serviceAccount string) {
```
We no longer need the `autoMount` flag, we can hardcode this to `false` in the builder.

Reply:
For Galera, we are explicitly mounting it here:
https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/builder/statefulset_builder.go#L197